src/bridge/SequencerInbox.sol

```diff
@@ -712,6 +946,67 @@ contract SequencerInbox is DelegateCallAware, GasRefundEnabled, ISequencerInbox
         return bridge.sequencerMessageCount();
     }
 
+    /// @dev    This is the `sync validity window` during which no proofs are required.
+    /// @notice Returns true if the inbox is in a synced state (no unexpected delays are possible)
+    function isSynced() internal view returns (bool) {
+        return (block.number <= buffer.syncExpiryBlockNumber &&
+            block.timestamp <= buffer.syncExpiryTimestamp);
+    }
+
+    /// @inheritdoc ISequencerInbox
+    function forceInclusionDeadline(uint64 blockNumber, uint64 timestamp)
+        external
+        view
+        returns (uint64, uint64)
+    {
+        (uint64 bufferBlocks, uint64 bufferSeconds) = buffer.pendingDelay(
+            blockNumber,
+            timestamp,
+            bufferConfig.thresholdBlocks,
+            bufferConfig.thresholdSeconds
+        );
+        (uint64 _delayBlocks, , uint64 _delaySeconds, ) = maxTimeVariationBufferable(
+            bufferBlocks,
+            bufferSeconds
+        );
+        return (blockNumber + _delayBlocks, timestamp + _delaySeconds);
+    }
+
+    function _setBufferConfig(BufferConfig memory bufferConfig_) internal {
+        if (!isDelayBufferable) revert NotDelayBufferable();
+        if (!DelayBuffer.isValidBufferConfig(bufferConfig_)) {
+            revert BadBufferConfig();
+        }
+
+        uint64 syncBlockNumber = buffer.syncExpiryBlockNumber > 0 &&
+            buffer.syncExpiryBlockNumber != type(uint64).max
+            ? buffer.syncExpiryBlockNumber - bufferConfig.thresholdBlocks
+            : 0;
+        uint64 syncTimestamp = buffer.syncExpiryTimestamp > 0 &&
+            buffer.syncExpiryBlockNumber != type(uint64).max
```

Comment:
when the threshold changes, the sync expiry calculation must be updated since it may no longer be valid (threshold dropped from 10 hours to 1 hour, the outdated sync expiry may be valid for the next 9hr 30min, but we should adjust this to a 30 min validity)

essentially this calculation recovers the original timestamp / blocknumber used to calculate the sync validity in the first place.